encoding/encoding.go

```diff
@@ -73,6 +75,7 @@ var registeredCompressor = make(map[string]Compressor)
 // registered with the same name, the one registered last will take effect.
 func RegisterCompressor(c Compressor) {
 	registeredCompressor[c.Name()] = c
+	grpcutil.RegisterCompressorName(c.Name())
```

Comment:
instead of this, how about something simpler:

```go
package grpc/internal

var RegisteredCompressorNames []string

...

package encoding

func RegisterCompressor(c Compressor) {
	registeredCompressor[c.Name()] = c
	internal.RegisteredCompressorNames = append(internal.RegisteredCompressorNames, c.Name())
}
```